toolbox/library/src/OSQP.cpp

```diff
@@ -318,14 +325,21 @@ bool wbt::block::OSQP::solverInitialization(const BlockInformation*blockInfo)
     // Setup options
     pImpl->sqSolver->settings()->setVerbosity(false);
 
-    // Set adaptive_rho to false to avoid converge problems
+    // Setup adaptive_rho
+    // Note that enabling adaptive_rho could cause converge problems
     // See https://github.com/oxfordcontrol/osqp/issues/151
-    pImpl->sqSolver->settings()->setAdaptiveRho(false);
```

Comment:
Interesting! I did not notice that adaptive_rho was already set to false, so the OSQP block should already be deterministic, or at least more deterministic w.r.t. to the use of OSQP with adaptive_rho enable (see https://github.com/google/osqp-cpp#faq).